robocode.battle/src/main/java/net/sf/robocode/battle/Battle.java

```diff
@@ -727,17 +728,17 @@ private void computeInitialPositions(String initialPositions) {
 
 			if (len >= 1 && coords[0].trim().length() > 0) {
 				try {
-					x = Double.parseDouble(coords[0].replaceAll("[^0-9.]", ""));
+					x = Double.parseDouble(coords[0].replaceAll(invalidCharacters, ""));
```

Comment:
While this resolves the SQ issue by reusing the regex string, another solution I'd suggest is encapsulating the string replacement behavior into a well name method and reusing that.
```java
private String stripNonNumericCharacters(String input) {
    return input.replaceAll("[^0-9.]", "");
}
```
```suggestion
					x = Double.parseDouble(stripNonNumericCharacters(coords[0]));
```